substrate/client/network/sync/src/strategy/state.rs

```diff
@@ -354,12 +350,31 @@ impl<B: BlockT> StateStrategy<B> {
 
 	/// Get actions that should be performed by the owner on [`WarpSync`]'s behalf
 	#[must_use]
-	pub fn actions(&mut self) -> impl Iterator<Item = StateStrategyAction<B>> {
+	pub fn actions(
+		&mut self,
+		network_service: &NetworkServiceHandle,
+	) -> impl Iterator<Item = SyncingAction<B>> {
 		let state_request = self.state_request().into_iter().map(|(peer_id, request)| {
-			StateStrategyAction::SendStateRequest {
+			let (tx, rx) = oneshot::channel();
+
+			network_service.start_request(
+				peer_id,
+				self.protocol_name.clone(),
+				request.encode_to_vec(),
+				tx,
+				IfDisconnected::ImmediateError,
```

Comment:
dq: Should we try to establish connection instead of immediate errors?

Would you think it is beneficial for strategies to provide this option in their `actions`?

Reply:
This is the same as it was before, so I didn't change the behavior here and so far there was no need to do this.

Maybe the assumption is that we only send requests to connected peers, so if someone disconnected we just error instead of attempting to dial them?